include/gamate.h

```diff
@@ -0,0 +1,200 @@
+ probably out=!bit16, bit0=bit16 xor bit13; runs through, so start value anything than 0
+8: 1st volume: 0..3 square volume; bit 4 envelope (higher priority)
+9: 2nd volume
+10: 3rd volume
+11/12: envelope delay time 0 fast, 0xffff slow/nearly no effect (2 22us, 4 56us)
+ frequency $800, envelope $10 2 times in pulse high time (4*16*16)
+13: envelope control
+ 0-3 one time falling
+ 4-7 one time rising
+ 8 falling
+ 9 one time falling
+ a starts with down falling, rising; same falling time, but double/longer cycle
+ b one time falling, on
+ c rising
+ d one time rising, on
+ e rising, falling (double cycle before restart)
+ f one time rising
+ bit 0: once only
+ bit 1: full tone
+ bit 2: start rising (else falling)
+ bit 3:
+*/
+
+#define JOY_DATA        0x4400
+
+#define JOY_DATA_UP     0x01
+#define JOY_DATA_DOWN   0x02
+#define JOY_DATA_LEFT   0x04
+#define JOY_DATA_RIGHT  0x08
+#define JOY_DATA_FIRE_A 0x10
+#define JOY_DATA_FIRE_B 0x20
+#define JOY_DATA_START  0x40
+#define JOY_DATA_SELECT 0x80
+
+/* LCD
+
+    resolution 160x152 in 4 greys/greens
+    2 256x256 sized bitplanes (2x 8kbyte ram)
+*/
+#define LCD_BASE        0x5000
+
+#define LCD_MODE        0x5001
+/*
+ bit 3..0 (from zeropage 15)
+        bit 0 set no normal screen display, seldom scrolling effects on screen;
+        bytes written to somewhat actual display refresh position!?
+    bytes read "random"
+  bit 1,2,3 no effect
+ bit 4 swaps plane intensity
+ bit 5 ? display effect
+ bit 6 on y auto increment (else auto x increment), reading
+ bit 7 ? lcd flickering
+*/
+#define LCD_MODE_INC_Y  0x40
+
+#define LCD_XPOS        0x5002  /* smooth scrolling X */
+#define LCD_YPOS        0x5003  /* smooth scrolling Y */
+/*
+ smooth scrolling until $c8 with 200 limit
+ after 200 display if ((value & 0xf) < 8) display of (value & 0xf) - 8
+ chaos lines from value + current line from plane 2 only then lines starting
+ with zero (problematic 200 limit/overrun implementation!?)
+*/
+#define LCD_X           0x5004  /* x-addr */
+/*
+ bit 5,6 no effect
+ bit 7 0 1st/1 2nd bitplane
+*/
+#define LCD_XPOS_PLANE1 0x00
+#define LCD_XPOS_PLANE2 0x80
+
+#define LCD_Y           0x5005  /* y-addr */
+
+#define LCD_READ        0x5006  /* read from RAM (no auto inc?) */
+#define LCD_DATA        0x5007  /* write to RAM */
+
+/* BIOS zeropage usage */
+
+/* locations 0x0a-0x0c, 0x0e-0x11 and 0xe8 are in use by the BIOS IRQ/NMI handlers */
+#define ZP_NMI_4800     0x0a    /* content of I/O reg 4800 gets copied here each NMI */
+
+#define ZP_IRQ_COUNT    0x0b    /* increments once per IRQ, used elsewhere in the
+                                   BIOS for synchronisation purposes */
+#define ZP_IRQ_CTRL     0x0c    /* if 0 then cartridge irq stubs will not get called */
+
+/* each of the following 4 increments by 1 per IRQ - it is _not_ a 32bit
+   counter (see code at $ffa6 in BIOS)
+   these are not used elsewhere in the bios and can be (re)setted as needed by
```

Comment:
"set" falls into "irregular" category. Meaning "setted" -> "set"
set 1 |sɛt|
verb (sets, setting; past and past participle: set)